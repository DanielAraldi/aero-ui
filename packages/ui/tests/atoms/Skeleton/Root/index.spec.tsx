import { Easing, Text } from 'react-native';
import { render, screen } from '@testing-library/react-native';

import { Skeleton, useSkeleton } from '../../../../';

interface SutTypes {
  childStub: jest.Mock<JSX.Element, [], any>;
}

const ChildStub = jest.fn(() => {
  const { activated, duration, useNativeDriver, easing } = useSkeleton();

  return (
    <>
      <Text testID='activated'>{String(activated)}</Text>
      <Text testID='duration'>{String(duration)}</Text>
      <Text testID='use-native-driver'>{String(useNativeDriver)}</Text>
      <Text testID='easing'>{String(easing)}</Text>
    </>
  );
});

function makeSut(): SutTypes {
  return {
    childStub: ChildStub,
  };
}

describe('<Skeleton.Root />', () => {
  it('Should render Skeleton.Root component', () => {
    render(<Skeleton.Root />);
    expect(screen.getByTestId('root')).toBeOnTheScreen();
  });

  it('Should pass props to the children correctly', () => {
    const { childStub: ChildStub } = makeSut();

    render(
      <Skeleton.Root>
        <ChildStub />
      </Skeleton.Root>
    );

    expect(ChildStub).toHaveBeenCalled();
    expect(screen.getByTestId('duration')).toHaveTextContent('2000');
    expect(screen.getByTestId('activated')).toHaveTextContent('true');
    expect(screen.getByTestId('use-native-driver')).toHaveTextContent('true');
    expect(screen.getByTestId('easing')).toHaveTextContent(
      String(Easing.bezier(0.4, 0, 0.6, 1))
    );
  });

  it('Should take a snapshot of the Skeleton.Root component', () => {
    const { childStub: ChildStub } = makeSut();

    const component = render(
      <Skeleton.Root>
        <ChildStub />
      </Skeleton.Root>
    );
    expect(component).toMatchSnapshot();
  });
});
