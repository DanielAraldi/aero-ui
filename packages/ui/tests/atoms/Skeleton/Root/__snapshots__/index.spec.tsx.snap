// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton.Root /> Should take a snapshot of the Skeleton.Root component 1`] = `
<View
  testID="root"
>
  <Text
    testID="activated"
  >
    true
  </Text>
  <Text
    testID="duration"
  >
    2000
  </Text>
  <Text
    testID="use-native-driver"
  >
    true
  </Text>
  <Text
    testID="easing"
  >
    function BezierEasing(x) {
    if (mX1 === mY1 && mX2 === mY2) {
      return x;
    }
    if (x === 0) {
      return 0;
    }
    if (x === 1) {
      return 1;
    }
    return calcBezier(getTForX(x), mY1, mY2);
  }
  </Text>
</View>
`;
